2.4/root/usr/bin/run-mongod-pet

```diff
@@ -0,0 +1,68 @@
+#!/bin/bash
+#
+# Run mongod as part of PetSet.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+source ${CONTAINER_SCRIPTS_PATH}/common.sh
+
+function usage() {
+  echo "You must specify the following environment variables:"
+  echo "  MONGODB_USER"
+  echo "  MONGODB_PASSWORD"
+  echo "  MONGODB_DATABASE"
+  echo "  MONGODB_ADMIN_PASSWORD"
+  echo "  MONGODB_KEYFILE_VALUE"
+  echo "  MONGODB_REPLICA_NAME"
+  echo "Optional variables:"
+  echo "  MONGODB_SERVICE_NAME (default: mongodb)"
+  echo "MongoDB settings:"
+  echo "  MONGODB_NOPREALLOC (default: true)"
+  echo "  MONGODB_SMALLFILES (default: true)"
+  echo "  MONGODB_QUIET (default: true)"
+  exit 1
+}
+
+# Make sure env variables don't propagate to mongod process.
+function unset_env_vars() {
+  unset MONGODB_USER MONGODB_PASSWORD MONGODB_DATABASE MONGODB_ADMIN_PASSWORD
+}
+
+function cleanup() {
+  echo "=> Shutting down MongoDB server ..."
+  if pgrep mongod; then
+    pkill -2 mongod
+  fi
+  wait_for_mongo_down
+  exit 0
+}
+
+trap 'cleanup' SIGINT SIGTERM
+
+# If user provides own config file use it and do not generate new one
+if [ ! -s $MONGODB_CONFIG_PATH ]; then
+  # Generate config file for MongoDB
+  envsubst < ${CONTAINER_SCRIPTS_PATH}/mongodb.conf.template > $MONGODB_CONFIG_PATH
+fi
+
+mongo_common_args="-f $MONGODB_CONFIG_PATH"
+if ! [[ -v MONGODB_USER && -v MONGODB_PASSWORD && -v MONGODB_DATABASE && -v MONGODB_ADMIN_PASSWORD && -v MONGODB_REPLICA_NAME && -v MONGODB_KEYFILE_VALUE ]]; then
+    usage
+fi
+
+# Run the MongoDB in 'replicated' mode
+setup_keyfile
+
+# Run the supervisor in background to manage replset
+${CONTAINER_SCRIPTS_PATH}/replset_petset_supervisor.sh "${1:-}" &
+
+# Run `unset_env_vars` and `mongod` in a subshell because
+# MONGODB_ADMIN_PASSWORD should still be defined when the trapped call to
+# `cleanup` references it.
+(
+  unset_env_vars
+  mongod $mongo_common_args --replSet "${MONGODB_REPLICA_NAME}"
```

Comment:
Good question and idea. Before we couldn't but now we I don't see a reason of not doing it.